src/asl.ts

```diff
@@ -897,10 +897,31 @@ export class ASL {
         const ref = expr.expr.ref();
         if (isIntegration<Integration>(ref)) {
           const serviceCall = new IntegrationImpl(ref);
-          const taskState = <State>{
-            ...serviceCall.asl(expr, this),
-            ...props,
-          };
+          const partialTask = serviceCall.asl(expr, this);
+
+          const { End, Next } = props;
+
+          const taskState = ((): State => {
+            switch (partialTask.Type) {
+              case "Wait":
+                return {
+                  ...(partialTask as Omit<Wait, "Next">),
+                  ...{ End, Next },
+                };
+              case "Choice":
+              case "Fail":
+              case "Succeed":
+                return partialTask as Choice | Fail | Succeed;
+              case "Task":
+              case "Parallel":
+              case "Pass":
+              case "Map":
+                return {
+                  ...partialTask,
+                  ...props,
+                } as Task | ParallelTask | Pass | MapTask;
+            }
```

Comment:
style: Please no switch - it's so fugly.

Reply:
I don't understand what this code is doing. What are these cases that need to be handled?

Reply:
Uhgg, but the `if else` was so verbose. Fine.

when `ResultPath` was `null`, on the `Wait` the deployment was failing. So this only injects the allowed properties into the nodes returned by the integrations.

some combination of ResultPath, End, Next based on what the state can handle.

Reply:
Perhaps add some comments because it's not obvious.

Reply:
:+1: